src/gui/editors/AutomationEditor.cpp

```diff
@@ -740,91 +738,85 @@ void AutomationEditor::mouseMoveEvent(QMouseEvent * mouseEvent )
 		return;
 	}
 
+	// If the mouse y position is inside the Automation Editor viewport
 	if( mouseEvent->y() > TOP_MARGIN )
 	{
 		float level = getLevel( mouseEvent->y() );
-		int x = mouseEvent->x();
+		// Get the viewport X position where the mouse is at
+		int x = mouseEvent->x() - VALUES_WIDTH;
 
-		x -= VALUES_WIDTH;
+		// If we are moving a value, we account for the offset from the node we
+		// had when clicking it in the first place (user might not have clicked exactly
+		// in the middle of the node)
 		if( m_action == MOVE_VALUE )
 		{
 			x -= m_moveXOffset;
 		}
 
-		int pos_ticks = x * MidiTime::ticksPerBar() / m_ppb +
-							m_currentPosition;
-		// m_mouseDownLeft used to prevent drag when drawing line
+		// Get the X position in ticks
+		int posTicks = (x * MidiTime::ticksPerBar()/m_ppb) + m_currentPosition;
+
+		// If we are in DRAW mode and the left mouse is down
 		if (m_mouseDownLeft && m_editMode == DRAW)
 		{
 			if( m_action == MOVE_VALUE )
 			{
-				// moving value
-				if( pos_ticks < 0 )
+				// If we moved the mouse past the beginning correct the position in ticks
+				if( posTicks < 0 )
```

Comment:
Can't suggest over deleted lines, but replaces the whole `if`.
```suggestion
				posTicks = qMax(posTicks, 0);
```

Reply:
There are two other places that weren't deleted where this happened, I changed it there!